src/main/java/ua/goit/dao/source/Data.java

```diff
@@ -0,0 +1,60 @@
+package ua.goit.dao.source;
+
+import ua.goit.model.Category;
+import ua.goit.model.Project;
+
+import static ua.goit.dao.source.Constants.*;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Data {
+  private static boolean init = false;
```

Comment:
I recommend you drop lazy initialization here and read up everything in var initialization e..g categorydata = initData(...)
alternatively fix the race condition which is available here :)